File: selfstudy/golang/apples_max.go
/*
task 202:
Apples - Max
You need to find the maximum value among apple weights in grams
*/

package main

import (
	"fmt"
	"reflect"
)

func HighestApples(apples []int) int {
	if len(apples) == 0 { return 0 }
	m := apples[0]
	for _, v := range apples {
		if v > m { m = v }
	}
	return m
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, HighestApples([]int{10, 20, 30})},
		{[]int{50, 60, 70}, HighestApples([]int{50, 60, 70})},
		{[]int{1, 2, 3}, HighestApples([]int{1, 2, 3})},
		{[]int{100, 200}, HighestApples([]int{100, 200})},
		{[]int{}, HighestApples([]int{})},
	}
	
	for idx, t := range tests {
		res := HighestApples(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// finally works

File: selfstudy/python/apples_max.py
'''
task 102:
Apples - Max
You need to find the maximum value among apple weights in grams
'''

def highest_apples(apples):
    return max(apples) if apples else 0

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], highest_apples([10, 20, 30])),
        ([50, 60, 70], highest_apples([50, 60, 70])),
        ([1, 2, 3], highest_apples([1, 2, 3])),
        ([100, 200], highest_apples([100, 200])),
        ([], highest_apples([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = highest_apples(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# works i think
